test/methods/equals.jl

```diff
@@ -0,0 +1,104 @@
+@testset "Points/MultiPoints" begin
+    p1 = LG.Point([0.0, 0.0])
+    p2 = LG.Point([0.0, 1.0])
+    # Same points
+    @test GO.equals(p1, p1)
+    @test GO.equals(p2, p2)
+    # Different points
+    @test !GO.equals(p1, p2)
+
+    mp1 = LG.MultiPoint([[0.0, 1.0], [2.0, 2.0]])
+    mp2 = LG.MultiPoint([[0.0, 1.0], [2.0, 2.0], [3.0, 3.0]])
+    # Same points
+    @test LG.equals(mp1, mp1)
+    @test LG.equals(mp2, mp2)
+    # Different points
+    @test !LG.equals(mp1, mp2)
+    @test !LG.equals(mp1, p1)
+end
+
+@testset "Lines/Rings" begin
+    l1 = LG.LineString([[0.0, 0.0], [0.0, 10.0]])
+    l2 = LG.LineString([[0.0, -10.0], [0.0, 20.0]])
+    # Equal lines
+    @test LG.equals(l1, l1)
+    @test LG.equals(l2, l2)
+    # Different lines
+    @test !LG.equals(l1, l2) && !LG.equals(l2, l1)
+
+    r1 = LG.LinearRing([[0.0, 0.0], [5.0, 5.0], [10.0, 0.0], [5.0, -5.0], [0.0, 0.0]])
+    r2 = LG.LinearRing([[3.0, 0.0], [8.0, 5.0], [13.0, 0.0], [8.0, -5.0], [3.0, 0.0]])
+    l3 = LG.LineString([[3.0, 0.0], [8.0, 5.0], [13.0, 0.0], [8.0, -5.0], [3.0, 0.0]])
+    # Equal rings
+    @test GO.equals(r1, r1)
+    @test GO.equals(r2, r2)
+    # Different rings
+    @test !GO.equals(r1, r2) && !GO.equals(r2, r1)
+    # Equal linear ring and line string
+    @test !GO.equals(r2, l3) # TODO: should these be equal?
```

Comment:
If we have two curves, but one is a linear ring and one is a line string , but they have all the same points, should they be the same? I don't think so, but maybe...

Reply:
I have determined that these should be equal given LibGEOS outputs. Similarly, a multi-polygon that holds a single polygon is equal to that same polygon. So it is about what the shape contains, not about the shape type, although obviously, some types preclude equality.